source/common/config/grpc_subscription_impl.h

```diff
@@ -1,49 +1,68 @@
+  // is_aggregated: whether our GrpcMux is also providing ADS to other Subscriptions, or whether
+  // it's all ours. The practical difference is that we ourselves must call start() on it only if
+  // we are the sole owner.
+  GrpcSubscriptionImpl(GrpcMuxSharedPtr grpc_mux, absl::string_view type_url,
                        SubscriptionCallbacks& callbacks, SubscriptionStats stats,
-                       Stats::Scope& scope, const RateLimitSettings& rate_limit_settings,
-                       std::chrono::milliseconds init_fetch_timeout, bool skip_subsequent_node)
-      : callbacks_(callbacks),
-        grpc_mux_(std::make_shared<Config::GrpcMuxImpl>(local_info, std::move(async_client),
-                                                        dispatcher, service_method, random, scope,
-                                                        rate_limit_settings, skip_subsequent_node)),
-        grpc_mux_subscription_(grpc_mux_, callbacks_, stats, type_url, dispatcher,
-                               init_fetch_timeout) {}
+                       std::chrono::milliseconds init_fetch_timeout, bool is_aggregated);
+  ~GrpcSubscriptionImpl() override;
+
+  void pause();
+  void resume();
 
   // Config::Subscription
-  void start(const std::set<std::string>& resource_names) override {
-    // Subscribe first, so we get failure callbacks if grpc_mux_.start() fails.
-    grpc_mux_subscription_.start(resource_names);
-    grpc_mux_->start();
-  }
+  void start(const std::set<std::string>& resource_names) override;
+  void updateResourceInterest(const std::set<std::string>& update_to_these_names) override;
 
-  void updateResourceInterest(const std::set<std::string>& update_to_these_names) override {
-    grpc_mux_subscription_.updateResourceInterest(update_to_these_names);
-  }
+  // Config::SubscriptionCallbacks (all pass through to callbacks_!)
+  void onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>& resources,
+                      const std::string& version_info) override;
+  void onConfigUpdate(const Protobuf::RepeatedPtrField<envoy::api::v2::Resource>& added_resources,
+                      const Protobuf::RepeatedPtrField<std::string>& removed_resources,
+                      const std::string& system_version_info) override;
+  void onConfigUpdateFailed(ConfigUpdateFailureReason reason, const EnvoyException* e) override;
+  std::string resourceName(const ProtobufWkt::Any& resource) override;
 
-  std::shared_ptr<Config::GrpcMuxImpl> grpcMux() { return grpc_mux_; }
+  GrpcMuxSharedPtr getGrpcMuxForTest() { return grpc_mux_; }
 
 private:
-  Config::SubscriptionCallbacks& callbacks_;
-  std::shared_ptr<Config::GrpcMuxImpl> grpc_mux_;
-  GrpcMuxSubscriptionImpl grpc_mux_subscription_;
+  GrpcMuxSharedPtr const grpc_mux_;
+  const std::string type_url_;
+  SubscriptionCallbacks& callbacks_;
+  SubscriptionStats stats_;
+  // NOTE: if another subscription of the same type_url has already been started, this value will be
```

Comment:
The comment doesn't match. Maybe move to type_url_?

Reply:
The comment is referring to init_fetch_timeout_, which does in fact get ignored after the very first GrpcSubscription has attached to the GrpcMux. Clarified the comment.